code/controllers/Processes/timer.dm

```diff
@@ -0,0 +1,77 @@
+var/global/datum/controller/process/timer/PStimer
+
+/datum/controller/process/timer
+	var/list/processing_timers = list()
+	var/list/hashes = list()
+
+/datum/controller/process/timer/setup()
+	name = "timer"
+	schedule_interval = 5 //every 0.5 seconds
+	PStimer = src
+
+/datum/controller/process/timer/statProcess()
+	..()
+	stat(null, "[processing_timers.len] timers")
+
+/datum/controller/process/timer/doWork()
+	if(!processing_timers.len)
+		disabled = 1 //nothing to do, lets stop firing.
+		return
+	for(var/datum/timedevent/event in processing_timers)
+		if(!event.thingToCall || qdeleted(event.thingToCall))
+			qdel(event)
+		if(event.timeToRun <= world.time)
+			spawn(-1)
```

Comment:
at /tg/ we changed this to call a proc using set waitfor = 0 so that it has no overhead if the called proc doesn't sleep